File: Python/1094 - Experiências.py
N = int(input())

total_coelhos = 0
total_ratos = 0
total_sapos = 0

for _ in range(N):
    quantidade, tipo = input().split()
    quantidade = int(quantidade)

    if tipo == "C":
        total_coelhos += quantidade
    elif tipo == "R":
        total_ratos += quantidade
    elif tipo == "S":
        total_sapos += quantidade

total_cobaias = total_coelhos + total_ratos + total_sapos

print(f"Total: {total_cobaias} cobaias")
print(f"Total de coelhos: {total_coelhos}")
print(f"Total de ratos: {total_ratos}")
print(f"Total de sapos: {total_sapos}")
print(f"Percentual de coelhos: {total_coelhos / total_cobaias * 100:.2f} %")
print(f"Percentual de ratos: {total_ratos / total_cobaias * 100:.2f} %")
print(f"Percentual de sapos: {total_sapos / total_cobaias * 100:.2f} %")

File: JavaScript/1094 - Experiências.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8').split('\n');
var N = parseInt(input[0]);

let total = 0, coelhos = 0, ratos = 0, sapos = 0;

for (let i = 1; i <= N; i++) {
    let [qtd, tipo] = input[i].split(' ');
    qtd = parseInt(qtd);
    total += qtd;

    if (tipo === 'C') coelhos += qtd;
    else if (tipo === 'R') ratos += qtd;
    else if (tipo === 'S') sapos += qtd;
}

console.log(`Total: ${total} cobaias`);
console.log(`Total de coelhos: ${coelhos}`);
console.log(`Total de ratos: ${ratos}`);
console.log(`Total de sapos: ${sapos}`);
console.log(`Percentual de coelhos: ${(coelhos * 100 / total).toFixed(2)} %`);
console.log(`Percentual de ratos: ${(ratos * 100 / total).toFixed(2)} %`);
console.log(`Percentual de sapos: ${(sapos * 100 / total).toFixed(2)} %`);